src/ftxui/component/renderer.cpp

```diff
@@ -69,6 +62,48 @@ Component Renderer(Component child, std::function<Element()> render) {
   return renderer;
 }
 
+/// @brief Return a focusable component, using |render| to render its interface.
+/// @param render The function drawing the interface, taking a boolean telling
+/// whether the component is focused or not.
+/// @ingroup component
+///
+/// ### Example
+///
+/// ```cpp
+/// auto screen = ScreenInteractive::TerminalOutput();
+/// auto renderer = Renderer([] (bool focused) {
+///   if (focused)
+///     return text("My interface") | inverted;
+///   else
+///     return text("My interface") | inverted;
+/// });
+/// screen.Loop(renderer);
+/// ```
+Component Renderer(std::function<Element(bool)> render) {
+  class Impl : public ComponentBase {
+   public:
+    Impl(std::function<Element(bool)> render) : render_(std::move(render)) {}
+
+   private:
+    Element Render() override { return render_(Focused()) | reflect(box_); }
+    bool Focusable() const override { return true; }
+    bool OnEvent(Event event) override {
+      if (event.is_mouse() && box_.Contain(event.mouse().x, event.mouse().y)) {
+        if (!CaptureMouse(event))
+          return false;
+
+        TakeFocus();
+      }
+
+      return false;
+    }
+    Box box_;
```

Comment:
The default constructor for Box doesn't initialize the members.
Consider initializing `box_` explicitly to avoid reads before initialization.

Ref: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-always

Reply:
Thanks!
(I pushed two patches together, I will fix this part elsewhere)